packages/core/src/StarknetWindowObject.ts

```diff
@@ -80,7 +254,6 @@ export interface IStarknetWindowObject {
   request: <T extends RpcMessage>(
     call: Omit<T, "result">,
   ) => Promise<T["result"]>
-  enable: (options?: { starknetVersion?: "v4" | "v5" }) => Promise<string[]>
```

Comment:
For dapps which uses the current version of get-starknet, will still call `enable` on the injected window object. 

To have backwards compatibility, we have two options:

1. Keep the interface same as current while only extending the `request` method.
2. Declare another type like:

```
import {
  ConnectedStarknetWindowObject,
  StarknetWindowObject,
} from "get-starknet-core"
import {
  StarknetWindowObject as OldStarknetWindowObject,
  ConnectedStarknetWindowObject as OldConnectedStarknetWindowObject,
} from "get-starknet-coreV3"

export type BackwardsCompatibleStarknetWindowObject = StarknetWindowObject &
  Pick<OldStarknetWindowObject, "account" | "provider" | "enable">

export type BackwardsCompatibleConnectedStarknetWindowObject =
  ConnectedStarknetWindowObject &
    Pick<OldConnectedStarknetWindowObject, "account" | "provider">
 ```
 
 I don't know if there is a better way around backward compatibility. Open to suggestions

Reply:
@dhruvkelawala @avimak 
I think that it will be safer to keep `enable` for the next 2 versions (for example) while logging a warning of deprecation on the client.
WDYT?

Reply:
@dhruvkelawala 
I don't see how backward compatibility type will assist us or prevent other consumers from using this package.
Keeping this `enable` definition will cause consumers implement it (and there is no reason to implement it if you implement "request") while removing it will not prevent others from running it.
The implementation of `enable` in get-starknet remains (while utilizing the `request` api).

@avimak 
I think it will be better to log a warning of deprecation in the `enable`.

WDYT?

Reply:
Here what I meant was, if a Wallet is using new StarknetWindowObject from get-starknet, it will not have `enable` method. Which means the dapps that use current get-starknet which calls `enable` will be broken. Wallets can provide backwards compatibility from their side but it means both wallets might do it in a non-standard way.

This is why I suggested extending the new StarknetWindowObject with a BackwardsCompatibleSWO and not add `enable` on the new SWO. This will be clear for the consumer, when to use which SWO.

Reply:
IMO it's better to have one implementation per version at the lib level, and wallets could import an old get-starknet lib in parallel for backward compatibility; pls remember that this API is also exposed to libs/dapps and having a wallet-internal mitigating interface might lead to confusion (while completely non-useful to them)